packages/datetime/src/dateInput.tsx

```diff
@@ -98,6 +98,12 @@ export interface IDateInputProps extends IDatePickerBaseProps, IProps {
     outOfRangeMessage?: string;
 
     /**
+     * Whether the date popover should be inline.
+     * @default true
+     */
+    popoverInline?: boolean;
```

Comment:
the thing is, we don't usually accept changes "to unblock in the short term," because we have to support it for much longer than that. i'm going to use this as an opportunity to investigate `popoverProps` because that's the preferred solution here.

also, @jchomicki doesn't seem to have configured circle yet so this PR can't even merge as is.